File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/Institution/Institution.cs
using System.Collections.Generic;

namespace Psi.Data.Models.ClientConfigurationModels.Institution
{
    public class Institution : SettingsBaseHelper
    {
        public Institution(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("Institution.PrivacyPolicy")]
        public string PrivacyPolicy
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.EnableContentSecurityPolicyHeaders")]
        public bool EnableContentSecurityPolicyHeaders
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.XFrameOptionsURLs")]
        public string XFrameOptionsURLs
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.ContentSecurityPolicy")]
        public string ContentSecurityPolicy
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.UtcOffset")]
        public double ServerUtcOffset
        {
            get => GetDoubleValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.EnableStrictTransportSecurityHeaders")]
        public bool EnableStrictTransportSecurityHeaders
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.StrictTransportSecurity.maxAge")]
        public string StrictTransportSecurityMaxAge
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.StrictTransportSecurity.includeSubdomains")]
        public string StrictTransportSecurityIncludeSubdomains
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.StrictTransportSecurity.Preload")]
        public string StrictTransportSecurityPreload
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("X.App.HomeBanking.InstitutionCultureID")]
        public string InstitutionCultureID
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("X.App.HomeBanking.TimeZone")]
        public string TimeZone
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("X.App.HomeBanking.EnableTimeZoneOffset")]
        public string EnableTimeZoneOffset
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("X.App.HomeBanking.TimeZoneOffset")]
        public string TimeZoneOffset
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.Admin.ScheduledTransfers.RestrictDatePickerStartDateHours")]
        public string RestrictDatePickerStartDateHours
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.Admin.ScheduledTransfers.DatePickerEarliestStartDateHour")]
        public int DatePickerEarliestStartDateHour
        {
            get => GetIntValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.Admin.BaseUrl")]
        public string AdminBaseUrl
        {
            get => GetValue();
            set => SetValue(value);
        }

	    [SettingKey("X.App.HomeBanking.gsAppURL")]
	    public string HomebankingBaseUrl
	    {
	        get => GetValue().TrimEnd('/');
	        set => SetValue(value);
	    }

        [SettingKey("Mobile.Institution.WebDomainWhitelist")]
        public List<string> WebDomainWhitelist
        {
            get => GetListValue();
            set => SetValue(value);
        }

        [SettingKey("X.App.HomeBanking.LogForwardedClientIP")]
        public bool LogForwardedClientIp
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("Institution.ShouldUseForwardedAuthHeader")]
        public bool ShouldUseForwardedAuthHeader
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("X.App.HomeBanking.IntraBankRoutingNumber")]
        public string RoutingNumber
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("X.App.HomeBanking.IntraBankName")]
        public string Name
        {
            get => GetValue();
            set => SetValue(value);
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/Institution/Institution.ts
// Generated imports
import { WebDomainWhitelist } from '../WebDomainWhitelist';

export interface Institution {
    /** @settingKey Institution.PrivacyPolicy */
    privacyPolicy: string;
    /** @settingKey Institution.EnableContentSecurityPolicyHeaders */
    enableContentSecurityPolicyHeaders: boolean;
    /** @settingKey Institution.XFrameOptionsURLs */
    xFrameOptionsURLs: string;
    /** @settingKey Institution.ContentSecurityPolicy */
    contentSecurityPolicy: string;
    /** @settingKey Institution.UtcOffset */
    serverUtcOffset: number;
    /** @settingKey Institution.EnableStrictTransportSecurityHeaders */
    enableStrictTransportSecurityHeaders: boolean;
    /** @settingKey Institution.StrictTransportSecurity.maxAge */
    strictTransportSecurityMaxAge: string;
    /** @settingKey Institution.StrictTransportSecurity.includeSubdomains */
    strictTransportSecurityIncludeSubdomains: string;
    /** @settingKey Institution.StrictTransportSecurity.Preload */
    strictTransportSecurityPreload: string;
    /** @settingKey X.App.HomeBanking.InstitutionCultureID */
    institutionCultureID: string;
    /** @settingKey X.App.HomeBanking.TimeZone */
    timeZone: string;
    /** @settingKey X.App.HomeBanking.EnableTimeZoneOffset */
    enableTimeZoneOffset: string;
    /** @settingKey X.App.HomeBanking.TimeZoneOffset */
    timeZoneOffset: string;
    /** @settingKey Institution.Admin.ScheduledTransfers.RestrictDatePickerStartDateHours */
    restrictDatePickerStartDateHours: string;
    /** @settingKey Institution.Admin.ScheduledTransfers.DatePickerEarliestStartDateHour */
    datePickerEarliestStartDateHour: number;
    /** @settingKey Institution.Admin.BaseUrl */
    adminBaseUrl: string;
    /** @settingKey X.App.HomeBanking.gsAppURL */
    homebankingBaseUrl: string;
    /** @settingKey Mobile.Institution.WebDomainWhitelist */
    list: WebDomainWhitelist;
    /** @settingKey X.App.HomeBanking.LogForwardedClientIP */
    logForwardedClientIp: boolean;
    /** @settingKey Institution.ShouldUseForwardedAuthHeader */
    shouldUseForwardedAuthHeader: boolean;
    /** @settingKey X.App.HomeBanking.IntraBankRoutingNumber */
    routingNumber: string;
    /** @settingKey X.App.HomeBanking.IntraBankName */
    name: string;
}
